store/s3/s3.go

```diff
@@ -0,0 +1,153 @@
+package s3
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/brunotm/replicant/store"
+	"github.com/brunotm/replicant/transaction"
+	"net/url"
+)
+
+var _ store.Store = (*Store)(nil)
+
+func init() {
+	store.Register("s3",
+		func(uri string) (s store.Store, err error) {
+			return New(uri)
+		})
+}
+
+type Store struct {
+	data       *s3.S3
+	bucketName string
+	prefix     string
+}
+
+// URI in the form of s3://<access>:<secret>/<bucket>/<path>
+func New(uri string) (*Store, error) {
+	//TODO
+	var err error
+	u, err := url.Parse(uri)
+	if err != nil {
+		return nil, err
```

Comment:
wrap the error with the %w directive for fmt.Errorf so the callee knows where it came from and can also compare the underlying error.